jme3-core/src/main/java/com/jme3/animation/SkeletonControl.java

```diff
@@ -443,9 +459,19 @@ public Node getAttachmentsNode(String boneName) {
                     + "in the skeleton.");
         }
 
-        Node n = b.getAttachmentsNode();
-        Node model = (Node) spatial;
-        model.attachChild(n);
+        updateTargetsAndMaterials(spatial);
+        Node n = b.getAttachmentsNode(targetGeometry);
```

Comment:
I assume if a model has animated geometries with different transforms, then animations will affect them differently and they won't fit together when animated. I suppose one could create a model with two unconnected parts, each having a distinct set of bones, like a pair of insects. In that case giving the bone a complete list of animated geometries might be helpful. But in the multi-geometry models I'm familiar with (including Elephant, Ninja, and Sinbad) all the geometries have identical transforms.